source/agora/consensus/protocol/Nominator.d

```diff
@@ -204,11 +210,11 @@ extern(D):
         this.ledger = ledger;
         this.enroll_man = enroll_man;
         this.store = new SCPEnvelopeStore(cacheDB);
-        // Create and stop timer immediately
-        this.envelope_timer = this.taskman.setTimer(EnvTaskDelay,
-            &this.envelopeProcessTask, Periodic.No);
-        this.envelope_timer.stop();
-        this.catchup_timer = catchup_timer;
+        // Create stopped timers
+        this.timers[TimersIdx.Envelope] = this.taskman.createTimer(&this.envelopeProcessTask);
+        this.timers[TimersIdx.Nomination] = this.taskman.createTimer(&this.checkNominate);
```

Comment:
Got confused, `handleSCPEnvelope` relies on this timer to be created within `startNominatingTimer` by checking `null`. This breaks that. If this is OK, then we shouldn't need setting this timer to `null` when stopped and recreating in `startNominatingTimer`, Vibe.d has [pending](https://vibed.org/api/vibe.core.core/Timer.pending).

Reply:
Again you are right, I should not create this here with the current logic of it being expected to be `null` until `startNominatingTimer` is called.

Reply:
I am not sure we can use `pending` as it is `false` when the `task` is actually running.

Reply:
For now I added a commit to correct this with explicit set to null and code comment to explain. In the future we can revisit this logic but it may mean an update to `Vibe.d` and `localrest`.

Reply:
Now refactored in https://github.com/bosagora/agora/pull/3056 using a different approach of checking if enrolled.